### ISSUE TITLE
Debian package

### ISSUE THREAD

**Issue by umlaeute** [libpd/libpd#53]
i'd be willing to package `libpd` for Debian (and derivatives like Ubuntu), but would like to ask first whether this makes actual sense.
- are there releases?
  - at least the git repository does not have any (release) tags at all.
- how stable is the API?
  - what are the "public" headers of libpd? `z_libpd.h` & `util/z_hook_util.h` or is there (planned to have) a `libpd.h` header?
  - judging from issue #44, i figure that there might still be some fundamental changes to the API (i don't know how fundamental the proper functioning of these hooks are in "real world")
  - since i only had a look at the low-level C-style API, how stable is the API for binding `foo`?
- how stable is the ABI?
  - for a Debian package i would package an ordinary shared system library (`libpd.so.1`).
  - it seems that as of now, this is _not_ the most common use-case of libpd. instead it seems that whoever uses libpd, just drops a static copy into their project, rather than using a shared library. this might make sense on restrictive platforms like `iOS`, though it's probably bad practice.
  - judging from a quick glance at the top-level Makefile, it seems that the build-system is totally ignorant of consistent versioning via [soname](http://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html). is this by intention?
- how do you test for regressions?
- upstream
  - how closely does libpd follow development of puredata? the current 'master' branch is still Pd-0.43-3 whereas Pd-vanilla currently has 0.45-4. i'm mainly asking because i think that `libpd` and `puredata` should be able to share the same set of externals (`dlopen()` is supported on Debian), which might become problematic if the versions diverge too much.

please do not get me wrong (some questions might sound a bit rude), i'm mainly asking to figure out whether it makes sense to create Debian packages
- is this something that some/many users would like to have
- is it feasible to maintain such a package on the long run

**Comment by danomatika** [comment]
Chris and I replaced the pure-data sources with a submodule from Miller's source forge git, so keeping track of new upstream versions should be easier at the expense of being able to add libpd-specfic things. At this point the project seems to have matured somewhat so I'm fine with the trade off.

**Comment by aalex** [comment]
I could probably help with the Debian / Ubuntu packaging, as well.

**Comment by umlaeute** [comment]
@aalex thanks for the offer.
however, the problem why libpd has not been packaged for Debian is not manpower, but having to maintain duplicate sources of Pd (one for the "puredata" package itself; and "libpd" requires a (potentially different) copy of the Pd sources as well).

**Comment by danomatika** [comment]
Note: I intend to change the build system to use the libpd z_* sources in `pure-data/src` for the next release.

**Comment by umlaeute** [comment]
oh i forgot: the Debian package of puredata also builds (and includes) a `libpd0` package since Pd-0.53 or so.
There hasn't been any Debian release since then, but it is available in trixie and onwards (and also in bookworm-backports):
- https://packages.debian.org/trixie/libpd0

**Comment by umlaeute** [comment]
ah, that `libpd0` package is of course only the C part of libpd.
that library is built from the Pd-sources (using the `autotools` build system)

No other language bindings have been packaged.

**Comment by danomatika** [comment]
> No other language bindings have been packaged (which i think _this_ issue is really about)

Yeah and it is a big issue... IMO this repo has grown too large and we keep getting more examples and frameworks and platform-specific package manager files... all of which I personally do not use nor are able to maintain. After release libpd 0.15, I would like to split things up into separate repos within the group and find maintainers for the various bindings etc.

Long story short: I would not propose making any language-binding packages pointing here right now.
